encoding/protobuf/protoc-gen-yarpc-go/internal/testing/testing.pb.yarpc.go

```diff
@@ -237,23 +242,489 @@ func (h *_SinkYARPCHandler) Fire(ctx context.Context, requestMessage proto.Messa
 	if requestMessage != nil {
 		request, ok = requestMessage.(*FireRequest)
 		if !ok {
-			return protobuf.CastError(emptySink_FireYARPCRequest, requestMessage)
+			return protobuf.CastError(emptySinkServiceFireYARPCRequest, requestMessage)
+		}
+	}
+	return h.server.Fire(ctx, request)
+}
+
+func newSinkServiceFireYARPCRequest() proto.Message {
+	return &FireRequest{}
+}
+
+func newSinkServiceFireYARPCResponse() proto.Message {
+	return &yarpcproto.Oneway{}
+}
+
+var (
+	emptySinkServiceFireYARPCRequest  = &FireRequest{}
+	emptySinkServiceFireYARPCResponse = &yarpcproto.Oneway{}
+)
+
+// AllYARPCClient is the YARPC client-side interface for the All service.
+type AllYARPCClient interface {
+	GetValue(context.Context, *GetValueRequest, ...yarpc.CallOption) (*GetValueResponse, error)
+	SetValue(context.Context, *SetValueRequest, ...yarpc.CallOption) (*SetValueResponse, error)
+	Fire(context.Context, *FireRequest, ...yarpc.CallOption) (yarpc.Ack, error)
+	HelloOne(context.Context, ...yarpc.CallOption) (AllServiceHelloOneYARPCClient, error)
+	HelloTwo(context.Context, *HelloRequest, ...yarpc.CallOption) (AllServiceHelloTwoYARPCClient, error)
+	HelloThree(context.Context, ...yarpc.CallOption) (AllServiceHelloThreeYARPCClient, error)
+}
+
+// AllServiceHelloOneYARPCClient sends HelloRequests and receives the single HelloResponse when sending is done.
+type AllServiceHelloOneYARPCClient interface {
+	Context() context.Context
+	Send(*HelloRequest) error
+	CloseAndRecv() (*HelloResponse, error)
+}
+
+// AllServiceHelloTwoYARPCClient receives HelloResponses, returning io.EOF when the stream is complete.
+type AllServiceHelloTwoYARPCClient interface {
+	Context() context.Context
+	Recv() (*HelloResponse, error)
+}
+
+// AllServiceHelloThreeYARPCClient sends HelloRequests and receives HelloResponses, returning io.EOF when the stream is complete.
+type AllServiceHelloThreeYARPCClient interface {
+	Context() context.Context
+	Send(*HelloRequest) error
+	Recv() (*HelloResponse, error)
```

Comment:
"open" generated code to accept contexts

Reply:
This comes at the cost of not matching the grpc-go API, and offline we discussed about how per-message options such as timeouts do not seem useful to me